crcmocks/rbac.py

```diff
@@ -1,33 +1,83 @@
-        {"resourceDefinitions": [], "permission": "inventory:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-    ],
+    "data": [],
 }
 
 
-# TODO: allow customizable rbac response per user
-@blueprint.route("/v1/access/")
+# TODO: support resource definitions?
+@blueprint.route("/v1/access/", methods=["GET"])
 def rbac_access():
+    application = request.args.get("application")
+    limit = int(request.args.get("limit", 100))
+    offset = int(request.args.get("offset", 0))
+    # get the identity header
+    identity_header = request.headers.get("X-Rh-Identity")
+
+    if not identity_header:
+        return "No identity header present in request\n", 401
+
+    # get the user
+    user, username, account_number = get_user_rh_identity(identity_header)
+
+    if not user:
+        return f"No user found in TinyDB with username:" \
+               f" {username} or account_number: {account_number}\n", 404
+    if len(user) > 1:
+        return f"Multiple users found with:" \
+               f" {username} or account_number: {account_number}\n", 406
+
+    # get the user permissions
+    if user[0].get("entitlements") is not None:
+        permissions = user[0]["permissions"].split(",")
+    else:
+        permissions = DEFAULT_PERMISSIONS.copy()
+
+    rbac_response = deepcopy(BASE_RBAC_RESPONSE)
+    # set links
+    rbac_response["links"]["first"] = f"/api/rbac/v1/access/?application={application or ''}" \
+                                      f"&format=json&limit={limit}" \
+                                      f"&offset={offset}"
+    rbac_response["links"]["last"] = f"/api/rbac/v1/access/?application={application or ''}" \
+                                     f"&format=json&limit={limit}" \
+                                     f"&offset={offset}"
+
+    limit_count = 1
+    # TODO: implement 'real' pagination
+    for i, perm in enumerate(permissions):
+        if i >= offset:
+            if application:
+                if perm.split(":")[0] in application:
+                    rbac_response["data"].append(
+                        {"resourceDefinitions": [], "permission": perm}
+                    )
+                    limit_count += 1
+            else:
+                rbac_response["data"].append(
+                    {"resourceDefinitions": [], "permission": perm}
+                )
+                limit_count += 1
+            if limit_count > limit:
```

Comment:
I *think* this will do one extra loop than expected. Maybe we should do `>=`?

Reply:
If I started `limit_count` at 0 that would be true, but I decided to start at `1` to avoid the `>=`, I can change it if you'd prefer it the other way around.

Reply:
Yeah I think that may read better, I had typed a whole comment stepping through the logic when it occured to me that once we're greater than the limit it will actually have ran the expected number of times. IE: If `limit_count` is 3, it ran twice. Which is odd to me I suppose, but I guess either way is a bit weird. I think I would personally prefer starting at 0 and doing just an `==` check. @bsquizz what do you think?

Reply:
Yeah I agree, much cleaner that way. Let me change this!